server/datastore/mysql/hosts.go

```diff
@@ -840,9 +850,13 @@ func filterHostsByMacOSSettingsStatus(sql string, opt fleet.HostListOptions, par
 	newSQL := ""
 	newParams := []interface{}{}
 
-	if opt.TeamFilter == nil || *opt.TeamFilter == 0 {
-		// add "no team" filter
-		newSQL += ` AND h.team_id IS NULL`
+	if opt.TeamFilter == nil {
```

Comment:
Not sure if I'm reviewing before the PR is actually ready, but in any case I think you wanted to have both `TeamFilter=nil` and `TeamFilter=0` result in filtering on "no team" when a macos setting filter is provided? (and remove the TODO below)

Reply:
We're handling `TeamFilter == 0` in `filterHostsByTeam` since that is the default behavior.

Reply:
Oh gotcha, I think it'd be good to mention that in the comment below since it happens at a distance.